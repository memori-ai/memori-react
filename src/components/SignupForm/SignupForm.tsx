import { User, Tenant } from '@memori.ai/memori-api-client/dist/types';
import React, { useEffect, useState } from 'react';
import Button from '../ui/Button';
import { useTranslation } from 'react-i18next';
import memoriApiClient from '@memori.ai/memori-api-client';
import toast from 'react-hot-toast';
import { mailRegEx, pwdRegEx, usernameRegEx } from '../../helpers/utils';
import { getErrori18nKey } from '../../helpers/error';

const reservedUserNames = [
  'memori',
  'private',
  'public',
  'api',
  '404',
  '500',
  'it',
  'en',
  'account',
  'auth',
  'logout',
  'privacy_and_cookie',
  'unauthorized',
  'users',
  'virtual_spaces',
];

export interface Props {
  tenant: Tenant;
  apiClient: ReturnType<typeof memoriApiClient>;
  onLogin: (user: User, token: string) => void;
  goToLogin: () => void;
  __TEST__waitingForOtp?: boolean;
}

const SignupForm = ({
  tenant,
  apiClient,
  onLogin,
  goToLogin,
  __TEST__waitingForOtp = false,
}: Props) => {
  const { t, i18n } = useTranslation();
  const lang = i18n.language === 'it' ? 'it' : 'en';

  const { userSignUp, userConfirmSignUp, resendVerificationCode } =
    apiClient.backend;

  const [email, setEmail] = useState<string>();
  const [username, setUsername] = useState<string>();
  const [password, setPassword] = useState<string>();
  const [confirmPassword, setConfirmPassword] = useState<string>();

  const pwdAcceptable = !password || (password && pwdRegEx.test(password));
  const pwdGreen = pwdAcceptable && password && password.length >= 24;
  const pwdEmpty = !password || password.length === 0;
  const pwdMeterValue =
    !pwdAcceptable || pwdEmpty
      ? 0
      : password.length < 8
      ? 15
      : password.length >= 32
      ? 100
      : (password.length - 8) * (50 / 24) + 50;

  const [loading, setLoading] = useState(false);
  const [loadingOtp, setLoadingOtp] = useState(false);
  const [waitingForOtp, setWaitingForOtp] = useState(__TEST__waitingForOtp);
  const [error, setError] = useState<string | null>(null);

  const [referral, setReferral] = useState<string>();
  useEffect(() => {
    setReferral(window.location.href);
  }, []);

  const generateUsernameFromEmail = (email: string) =>
    email.split('@')[0].replace(/[^a-zA-Z0-9]/g, '');

  const signup = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const form = e.currentTarget as HTMLFormElement;

    const tenantID = form.tenant.value ?? tenant.id;

    const eMail = (form.eMail.value ?? email).toLowerCase();
    const userName = form.userName.value ?? username;
    const password = form.password.value;

    const birthDate = new Date(form.birthDate?.value).toISOString();
    const tnCAndPPAccepted = form.tnCAndPPAccepted?.checked;
    const pAndCUAccepted = form.pAndCUAccepted?.checked;

    const referral = form.referral?.value;

    if (!eMail || !userName || !password || !birthDate || !tnCAndPPAccepted) {
      setError(t('missingData'));
      return;
    }

    setLoading(true);
    setError(null);

    const age =
      (new Date().getTime() - new Date(birthDate).getTime()) /
      1000 /
      60 /
      60 /
      24 /
      365;

    if (age < 14) {
      toast.error(t('login.underage', { age: 14 }));
      setError(t('login.underage', { age: 14 }));
      return;
    }

    try {
      const { user, ...resp } = await userSignUp({
        tenant: tenantID,
        eMail,
        userName,
        password,
        birthDate,
        tnCAndPPAccepted,
        tnCAndPPAcceptanceDate: new Date().toISOString(),
        pAndCUAccepted,
        pAndCUAcceptanceDate: new Date().toISOString(),
        referral,
      });
      if (resp.resultCode !== 0) {
        console.error(resp);
        toast.error(t(getErrori18nKey(resp.resultCode)));
      } else if (user) {
        toast.success(t('success'));
        setWaitingForOtp(true);
      }
    } catch (e) {
      let err = e as Error;
      console.error('[signup]', err);
      if (err?.message) toast.error(err.message);
    } finally {
      setLoading(false);
    }
  };

  const resendOtp = async (e: React.MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
    if (!username) return;

    setLoadingOtp(true);

    try {
      const resp = await resendVerificationCode({
        tenant: tenant?.id,
        userName: username,
      });

      if (resp.resultCode === 0) {
        toast.success(t('resentVerificationCode'));
      } else {
        console.error(resp);
        toast.error(t(getErrori18nKey(resp.resultCode)));
      }
    } catch (e) {
      let err = e as Error;
      console.error('[resend otp]', err);
      if (err?.message) toast.error(err.message);
    } finally {
      setLoadingOtp(false);
    }
  };

  const validateOtp = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const form = e.currentTarget as HTMLFormElement;

    const tenantID = form.tenant.value ?? tenant.id;
    const userName = form.userName.value ?? username;
    const pwd = form.password.value ?? password;

    const verificationCode = form.verificationCode.value;

    setLoading(true);
    setError(null);

    try {
      const { user, token, ...resp } = await userConfirmSignUp({
        tenant: tenantID,
        userName,
        password: pwd,
        verificationCode,
      });

      if (resp.resultCode !== 0) {
        console.error(resp);
        toast.error(t(getErrori18nKey(resp.resultCode)));
      } else if (user && token) {
        toast.success(t('success'));
        onLogin(user, token);
      }
    } catch (e) {
      let err = e as Error;
      console.error('[validate otp]', err);
      if (err?.message) toast.error(err.message);
    } finally {
      setLoading(false);
    }
  };

  return waitingForOtp ? (
    <>
      <form className="memori--login-drawer--form" onSubmit={validateOtp}>
        <input type="hidden" name="userName" value={username} />
        <input type="hidden" name="password" value={password} />
        <input type="hidden" name="tenant" value={tenant.id} />

        <label htmlFor="#verificationCode">
          {t('login.otpCode')}
          <input
            id="verificationCode"
            name="verificationCode"
            type="text"
            required
            autoComplete="one-time-code"
            placeholder={t('login.otpCode') || 'OTP'}
          />
        </label>
        <p>
          <Button outlined onClick={resendOtp} loading={loadingOtp}>
            {t('login.resendVerificationCode')}
          </Button>
        </p>

        <Button htmlType="submit" primary loading={loading}>
          {t('confirm')}
        </Button>
      </form>
      {error && (
        <p role="alert" className="memori--login-drawer--error">
          {error}
        </p>
      )}
    </>
  ) : (
    <>
      <form className="memori--login-drawer--form" onSubmit={signup}>
        <input type="hidden" name="tenant" value={tenant.id} />
        <input type="hidden" name="referral" value={referral} />

        <label htmlFor="#eMail">
          {t('login.email')}
          <input
            type="email"
            name="eMail"
            id="eMail"
            required
            autoComplete="email"
            placeholder={t('login.email') || 'Email'}
            onChange={e => setEmail(e.target.value)}
            aria-invalid={!!email?.length && !mailRegEx.test(email)}
          />
        </label>
        {!!email?.length && !mailRegEx.test(email) && (
          <p className="memori--login-drawer--inline-error">
            {t('login.emailFormatError')}
          </p>
        )}

        <label htmlFor="userName">
          {t('login.username')}
          <input
            type="text"
            name="userName"
            id="userName"
            required
            autoComplete="username"
            placeholder={t('login.username') || 'Username'}
            onFocus={() => {
              if (!username?.length && !!email?.length) {
                let username = generateUsernameFromEmail(email);
                const field = document.getElementById(
                  'userName'
                ) as HTMLInputElement | null;
                if (field) {
                  field.value = username;
                  setUsername(username);
                }
              }
            }}
            onChange={e => setUsername(e.target.value)}
            aria-invalid={
              !!username?.length &&
              (reservedUserNames.includes(username.toLowerCase()) ||
                !usernameRegEx.test(username))
            }
          />
        </label>
        {!!username?.length &&
          reservedUserNames.includes(username.toLowerCase()) && (
            <p className="memori--login-drawer--inline-error">
              {t('login.usernameContainsReservedWord')}
            </p>
          )}
        {!!username?.length && !usernameRegEx.test(username) && (
          <p className="memori--login-drawer--inline-error">
            {t('login.usernameFormatError')}
          </p>
        )}

        <label htmlFor="#birthDate">
          {t('login.birthDate')}
          <input
            id="birthDate"
            name="birthDate"
            type="date"
            required
            autoComplete="bday"
          />
        </label>
        <p>
          <small>{t('login.birthDateHelper')}</small>
        </p>

        <label htmlFor="#password">
          {t('login.password')}
          <input
            id="password"
            name="password"
            type="password"
            required
            autoComplete="new-password"
            placeholder={t('login.password') || 'Password'}
            onChange={e => setPassword(e.target.value)}
            aria-invalid={!pwdAcceptable}
          />
        </label>
        {!pwdAcceptable && (
          <p className="memori--login-drawer--inline-error">
            {t('login.passwordFormatError')}
          </p>
        )}

        <label htmlFor="#confirm-password">
          {t('login.confirmPassword')}
          <input
            id="confirm-password"
            name="confirmPassword"
            type="password"
            required
            autoComplete="new-password"
            placeholder={t('login.confirmPassword') || 'Password'}
            onChange={e => setConfirmPassword(e.target.value)}
            aria-invalid={
              !!password?.length &&
              !!confirmPassword?.length &&
              password !== confirmPassword
            }
          />
        </label>
        {!!password?.length &&
          !!confirmPassword?.length &&
          password !== confirmPassword && (
            <p className="memori--login-drawer--inline-error">
              {t('login.passwordMatchingError')}
            </p>
          )}

        <meter
          className="memori--login-drawer--password-meter"
          min={0}
          low={33}
          high={66}
          optimum={80}
          max={100}
          value={pwdMeterValue}
          id="password-strength-meter"
        />
        <small>
          {t(
            `login.pwd${
              pwdGreen ? 'Strong' : pwdAcceptable ? 'Acceptable' : 'Weak'
            }`
          )}
        </small>

        <label className="memori-checkbox">
          <span className="memori-checkbox--input-wrapper">
            <input
              type="checkbox"
              name="tnCAndPPAccepted"
              required
              className="memori-checkbox--input"
            />
            <span className="memori-checkbox--inner" />
          </span>
          <span className="memori-checkbox--text">
            {t('login.privacyLabel')}{' '}
            <a
              href={`https://memori.ai/${lang}/privacy_and_cookie`}
              target="_blank"
              rel="noopener noreferrer"
            >
              {t('login.privacyAndCookiePolicy')}
            </a>{' '}
            {t('login.and')}{' '}
            <a
              href={`https://memori.ai/${lang}/tos`}
              target="_blank"
              rel="noopener noreferrer"
            >
              {t('login.termsOfService')}
            </a>
          </span>
        </label>

        <label className="memori-checkbox">
          <span className="memori-checkbox--input-wrapper">
            <input
              type="checkbox"
              name="pAndCUAccepted"
              className="memori-checkbox--input"
            />
            <span className="memori-checkbox--inner" />
          </span>
          <span className="memori-checkbox--text">
            {t('login.pAndCUAccepted')}{' '}
            <small>
              <em>({t('login.optional')})</em>
            </small>
          </span>
        </label>
        <p>
          <small>{t('login.deepThoughtExplaination')}</small>
        </p>

        <Button htmlType="submit" primary loading={loading}>
          {t('login.signUp')}
        </Button>

        <p className="memori--login-drawer--signup">
          {t('login.alreadyHaveAnAccount')}{' '}
          <Button outlined onClick={goToLogin}>
            {t('login.backToLogin')}
          </Button>
        </p>
      </form>

      {error && (
        <p role="alert" className="memori--login-drawer--error">
          {error}
        </p>
      )}
    </>
  );
};

export default SignupForm;
