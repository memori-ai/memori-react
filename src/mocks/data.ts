import {
  Medium,
  Memori,
  Message,
  Memory,
  Integration,
  Tenant,
  DialogState,
  KnownFact,
  ExpertReference,
  User,
  Venue,
} from '@memori.ai/memori-api-client/dist/types';

export const sessionID = '131165be-9d1a-42fb-a3ce-e8f86d40c88f';

export const tenant: Tenant = {
  adminCount: 3,
  config: {
    name: 'AIsuru',
    requirePosition: false,
    showNewUser: true,
  },
  creationTimestamp: '2023-05-31T14:32:48.885287Z',
  description: 'AIsuru',
  disableRegistration: false,
  id: 'www.aisuru.com',
  lastChangeTimestamp: '2023-05-31T14:32:48.885287Z',
  logoURL: 'https://aisuru.com/images/aisuru/logo.png',
  maxAdmins: 0,
  maxCompletions: 0,
  maxCompletionsPerUser: 0,
  maxFreeSessions: 400,
  maxFreeSessionsPerUser: 100,
  maxMemoriPerAdmin: 0,
  maxMemoriPerUser: 3,
  maxTotalMemori: 0,
  maxUsers: 0,
  memoriCount: 0,
  name: 'www.aisuru.com',
  nonFreeSessionCost: 0.02,
  paying: true,
  tenantID: '96caa4b4-31a4-48e5-8163-dec61869a2a7',
  theme: 'aisuru',
  userCount: 0,
};

export const user: User = {
  tenant: 'localhost:3000',
  userID: '97c42d18-ffe4-47e1-a3c7-e42729f1e6a3',
  userName: 'nzambello',
  eMail: 'nicola@nzambello.dev',
  admin: false,
  maxMemori: 0,
  enableMemoriCreation: true,
  enableBoardOfExperts: true,
  maxFreeSessions: 0,
  tnCAndPPAccepted: true,
  tnCAndPPAcceptanceDate: '2021-03-01T00:00:00.000Z',
  pAndCUAccepted: true,
  pAndCUAcceptanceDate: '2021-03-01T00:00:00.000Z',
  birthDate: '1900-03-01T00:00:00.000Z',
  age: 28,
  avatarURL: 'https://avatars.githubusercontent.com/u/21101435?v=4',
};

export const memori: Memori = {
  memoriID: '25ced51c-3520-41af-8bbe-222d861b8e32',
  engineMemoriID: '66b4e161-2431-4b21-9b70-d8c27de730ca',
  name: 'Memori',
  memoriConfigurationID: 'MemoriCloud-it_IT',
  description: 'Lorem ipsum.',
  voiceType: 'male',
  isGiver: true,
  isReceiver: false,
  privacyType: 'PUBLIC',
  needsPosition: false,
  culture: 'it-IT',
  categories: [],
  publishedInTheMetaverse: true,
  exposed: true,
  enableCompletions: true,
  nsfw: false,
  ageRestriction: 14,
  contentQualityIndex: 66.6,
  contentQualityIndexTimestamp: '2021-03-01T12:00:00.000Z',
  ownerUserName: 'username',
  ownerTenantName: 'aisuru.com',
  ownerUserID: '97c42d18-ffe4-47e1-a3c7-e42729f1e6a3',
  metaverseEnvironment: 'synthwave',
  secretToken: 'awanagana',
  giverPIN: '1234',
  giverTag: 'üßë‚Äçüíª',
  avatarURL: 'https://aisuru.com/images/aisuru/square_logo.png',
  coverURL: 'https://aisuru.com/images/aisuru/og-image.png',
};

export const history: Message[] = [
  {
    text: "Ciao, io sono test, c'√® qualcosa che vorresti chiedermi?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Prova',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: "Mi dispiace, le mie risposte sono limitate. Devi farmi le domande giuste. C'√® altro che vuoi sapere?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Come faccio a fare delle cose con questa cosa?',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Ecco qui come.',
    media: [
      {
        mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efcd',
        url: 'https://memori.ai',
        mimeType: 'text/html',
        title: 'Link',
      },
    ],
    timestamp: '2021-03-01T12:00:00.000Z',
    contextVars: {
      TEST: 'test',
    },
  },
  {
    text: 'Ah, grazie! Ciao!',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Arrivederci.',
    timestamp: '2021-03-01T12:00:00.000Z',
    contextVars: {
      TEST: 'test',
    },
  },
];

export const historyWithMedia: Message[] = [
  {
    text: "Ciao, io sono test, c'√® qualcosa che vorresti chiedermi?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Prova',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: "Mi dispiace, le mie risposte sono limitate. Devi farmi le domande giuste. C'√® altro che vuoi sapere?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Come faccio a fare delle cose con questa cosa?',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Ecco qui delle cose per te.',
    media: [
      {
        mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efcd',
        url: 'https://memori.ai/en',
        mimeType: 'text/html',
        title: 'Link Memori Srl',
      },
      {
        mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efce',
        url: 'https://rawmaterial.it/en',
        mimeType: 'text/html',
        title: 'Link RawMaterial',
      },
      {
        mediumID: '95226d7e-7bae-465e-8b80-995587bb5971',
        mimeType: 'text/html',
        title: 'Introducing Plone Remix | Vimeo',
        url: 'https://vimeo.com/766468314',
      },
      {
        mediumID: '95226d7e-7bae-465e-8b80-995587bb5969',
        mimeType: 'text/html',
        title: 'A sustainable web: is it possible? - Nicola Zambello | YouTube',
        url: 'https://www.youtube.com/watch?v=feH26j3rBz8',
      },
      ...Array.from({ length: 3 }, (_, i) => ({
        mediumID: `95226d7e-7bae-465e-8b80-995587bb597${i}`,
        mimeType: 'image/png',
        title: `Image ${i}`,
        url: `https://picsum.photos/${i % 2 ? '200' : '300'}/${
          i % 3 ? '300' : '200'
        }?random=${i}`,
      })),
    ],
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Ah, grazie! Ciao!',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Arrivederci.',
    timestamp: '2021-03-01T12:00:00.000Z',
  },
];

export const historyWithAIGeneratedMessages: Message[] = [
  {
    text: "Ciao, io sono test, c'√® qualcosa che vorresti chiedermi?",
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Prova',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Sa. Sa. Prova',
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Come faccio a fare delle cose con questa cosa?',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Ecco qui come.',
    media: [
      {
        mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efcd',
        url: 'https://memori.ai',
        mimeType: 'text/html',
        title: 'Link',
      },
    ],
    timestamp: '2021-03-01T12:00:00.000Z',
    generatedByAI: true,
    contextVars: {
      TEST: 'test',
    },
  },
  {
    text: 'Ah, grazie! Ciao!',
    fromUser: true,
    timestamp: '2021-03-01T12:00:00.000Z',
  },
  {
    text: 'Arrivederci.',
    timestamp: '2021-03-01T12:00:00.000Z',
    contextVars: {
      TEST: 'test',
    },
  },
];

export const dialogState: DialogState = {
  state: 'R1',
  previousState: 'I0',
  stateName: 'WaitingForReceiverQuestion',
  confidence: 1,
  knownTags: {
    '‚ò†Ô∏è': 'test',
    'üòé': 'Ciccio',
  },
  emission: "Ciao, io sono test, c'√® qualcosa che vorresti chiedermi?",
  hints: ['Va bene', 'No grazie'],
  media: [],
  acceptsTimeout: true,
  acceptsAbort: false,
  acceptsMedia: false,
  acceptsDate: false,
  acceptsPlace: false,
  acceptsTag: false,
  giverID: 'c832e2dc-403c-4baf-a3b7-2374e100dbcf',
  contextVars: {},
};

export const medium: Medium = {
  mediumID: 'a196b513-d745-4121-8913-8f457b999add',
  url: 'https://api.lorem.space/image/game?w=150&h=220&hash=8B7BCDC2',
  mimeType: 'image/jpeg',
  title: 'Game Cover',
  creationTimestamp: '2022-05-03T08:57:30.584439Z',
  creationName: 'fcfda4a7-bb19-41eb-8384-db0d22f825ec',
  lastChangeTimestamp: '2022-05-03T14:19:09.932865Z',
  lastChangeName: 'fcfda4a7-bb19-41eb-8384-db0d22f825ec',
};

export const integration: Integration = {
  integrationID: 'cb3c4776-7f0b-4f97-a773-c32a5d7a3bf1',
  memoriID: '25ced51c-3520-41af-8bbe-222d861b8e32',
  type: 'LANDING_EXPERIENCE',
  state: 'NEW',
  deviceEmails: [],
  customData:
    '{"textColor":"#2a2a2a","buttonBgColor":"#823ce1","buttonTextColor":"#ffffff","name":"Web","globalBackground":"https://assets.memori.ai/api/v2/asset/364e498c-11da-42d5-9e32-19e5d137d4b8.jpeg","blurBackground":true,"innerBgColor":"light","innerBgAlpha":0.8,"multilanguage":true,"avatar":"readyplayerme","avatarURL":"https://assets.memori.ai/api/v2/asset/b791f77c-1a94-4272-829e-eca82fcc62b7.glb#1669663599444"}',
  resources: [],
  publish: true,
  creationTimestamp: '2022-06-11T14:13:45.685038Z',
  lastChangeTimestamp: '2022-06-11T14:13:45.685038Z',
};

export const knownFact: KnownFact = {
  knownFactID: 'b0b0b0b0-b0b0-b0b0-b0b0-b0b0b0b0b0b3',
  knownFactType: 'ShortTerm',
  text: 'I am a known fact. Quisque in ultrices lectus. Nulla at urna diam. Proin sodales lobortis libero eu facilisis.',
  creationTimestamp: '2023-12-01T13:40:25.235896Z',
  creationSessionID: '0ce713c0-c8f1-4aed-a2a6-40f81c06854f',
  lastChangeTimestamp: '2023-12-01T13:44:04.832072Z',
  lastChangeSessionID: '0ce713c0-c8f1-4aed-a2a6-40f81c06854f',
};

export const expertReference: ExpertReference = {
  expertID: 'f016f204-c307-483b-8891-680a3c974c53',
  name: 'TEST MEMORI',
  description: 'lui √® competente',
  default: true,
  expertMemoriID: 'dfc44f1c-1ba7-4e1e-a234-28aa8b6b3d32',
  expertBaseURL: 'http://localhost:7778',
  creationTimestamp: '2023-12-01T13:40:25.235896Z',
  creationSessionID: '0ce713c0-c8f1-4aed-a2a6-40f81c06854f',
  lastChangeTimestamp: '2023-12-01T13:44:04.832072Z',
  lastChangeSessionID: '0ce713c0-c8f1-4aed-a2a6-40f81c06854f',
};

export const venue: Venue = {
  latitude: 44.66579,
  longitude: 11.48823,
  placeName: 'Altedo, Bologna, Italy',
  uncertainty: 2,
};

export const memoryQuestion: Memory = {
  memoryID: '08c0697d-f7f3-4a47-9970-aff75f01fb6c',
  memoryType: 'Question',
  lastRead: undefined,
  readOccurrences: undefined,
  receiverID: undefined,
  receiverTag: undefined,
  receiverName: undefined,
  media: [
    {
      mediumID: 'c6851968-5d4d-409a-ae75-f22ec077efcd',
      mimeType: 'text/html',
      url: 'https://rawmaterial.it/it',
      title: 'RawMaterial',
    },
  ],
  title: 'sei umano',
  titleVariants: ['sei reale'],
  answers: [
    {
      text: 'se ti dico si ci credi?',
      preformatted: false,
      creationTimestamp: '2022-03-23T09:37:25.410377',
      creationName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
      lastChangeTimestamp: '2022-03-23T09:37:25.410406',
      lastChangeName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
    },
    {
      text: 'probabilmente lo sono pi√π di te',
      preformatted: false,
      creationTimestamp: '2022-03-23T09:37:36.957656',
      creationName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
      lastChangeTimestamp: '2022-03-23T09:37:36.957657',
      lastChangeName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
    },
  ],
  date: undefined,
  dateUncertaintyDays: undefined,
  placeName: undefined,
  placeLatitude: undefined,
  placeLongitude: undefined,
  placeUncertaintyKm: undefined,
  preformatted: false,
  conclusive: false,
  notPickable: false,
  contextVarsToSet: {
    SUBJECT: 'TRAVEL',
  },
  contextVarsToMatch: {
    SUBJECT: 'TRANSPORTS',
  },
  creationTimestamp: '2022-03-23T09:37:16.568149',
  creationName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
  lastChangeTimestamp: '2022-03-23T09:37:36.957695',
  lastChangeName: '6cfec2bd-a73f-4a21-b7ff-dd92d6db59c4',
};
