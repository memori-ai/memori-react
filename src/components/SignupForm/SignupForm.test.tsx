import React from 'react';
import { render } from '@testing-library/react';
import SignupForm from './SignupForm';
import { tenant } from '../../mocks/data';

beforeEach(() => {
  // @ts-ignore
  window.IntersectionObserver = jest.fn(() => ({
    observe: jest.fn(),
    unobserve: jest.fn(),
    disconnect: jest.fn(),
    takeRecords: jest.fn(),
  }));
});

it('renders SignupForm unchanged', () => {
  const { container } = render(
    <SignupForm
      apiUrl="https://backend.memori.ai"
      tenant={tenant}
      onLogin={jest.fn()}
      goToLogin={jest.fn()}
    />
  );
  expect(container).toMatchSnapshot();
});

it('renders SignupForm on otp form unchanged', () => {
  const { container } = render(
    <SignupForm
      apiUrl="https://backend.memori.ai"
      tenant={tenant}
      onLogin={jest.fn()}
      goToLogin={jest.fn()}
      __TEST__waitingForOtp
    />
  );
  expect(container).toMatchSnapshot();
});
