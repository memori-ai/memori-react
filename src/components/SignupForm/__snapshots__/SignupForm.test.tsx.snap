// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SignupForm on otp form unchanged 1`] = `
<div>
  <form
    class="memori--login-drawer--form"
  >
    <input
      name="userName"
      type="hidden"
      value=""
    />
    <input
      name="password"
      type="hidden"
      value=""
    />
    <input
      name="tenant"
      type="hidden"
      value="www.aisuru.com"
    />
    <label
      for="#verificationCode"
    >
      login.otpCode
      <input
        autocomplete="one-time-code"
        id="verificationCode"
        name="verificationCode"
        placeholder="login.otpCode"
        required=""
        type="text"
      />
    </label>
    <p>
      <button
        class="memori-button memori-button--outlined memori-button--rounded memori-button--padded"
      >
        login.resendVerificationCode
      </button>
    </p>
    <button
      class="memori-button memori-button--primary memori-button--rounded memori-button--padded"
      type="submit"
    >
      confirm
    </button>
  </form>
</div>
`;

exports[`renders SignupForm unchanged 1`] = `
<div>
  <form
    class="memori--login-drawer--form"
  >
    <input
      name="tenant"
      type="hidden"
      value="www.aisuru.com"
    />
    <input
      name="referral"
      type="hidden"
      value="http://localhost/"
    />
    <label
      for="#eMail"
    >
      login.email
      <input
        aria-invalid="false"
        autocomplete="email"
        id="eMail"
        name="eMail"
        placeholder="login.email"
        required=""
        type="email"
      />
    </label>
    <label
      for="userName"
    >
      login.username
      <input
        aria-invalid="false"
        autocomplete="username"
        id="userName"
        name="userName"
        placeholder="login.username"
        required=""
        type="text"
      />
    </label>
    <label
      for="#birthDate"
    >
      login.birthDate
      <input
        autocomplete="bday"
        id="birthDate"
        name="birthDate"
        required=""
        type="date"
      />
    </label>
    <p>
      <small>
        login.birthDateHelper
      </small>
    </p>
    <label
      for="#password"
    >
      login.password
      <input
        aria-invalid="false"
        autocomplete="new-password"
        id="password"
        name="password"
        placeholder="login.password"
        required=""
        type="password"
      />
    </label>
    <label
      for="#confirm-password"
    >
      login.confirmPassword
      <input
        aria-invalid="false"
        autocomplete="new-password"
        id="confirm-password"
        name="confirmPassword"
        placeholder="login.confirmPassword"
        required=""
        type="password"
      />
    </label>
    <meter
      class="memori--login-drawer--password-meter"
      high="66"
      id="password-strength-meter"
      low="33"
      max="100"
      min="0"
      optimum="80"
      value="0"
    />
    <small>
      login.pwdAcceptable
    </small>
    <label
      class="memori-checkbox"
    >
      <span
        class="memori-checkbox--input-wrapper"
      >
        <input
          class="memori-checkbox--input"
          name="tnCAndPPAccepted"
          required=""
          type="checkbox"
        />
        <span
          class="memori-checkbox--inner"
        />
      </span>
      <span
        class="memori-checkbox--text"
      >
        login.privacyLabel
         
        <a
          href="https://memori.ai/en/privacy_and_cookie"
          rel="noopener noreferrer"
          target="_blank"
        >
          login.privacyAndCookiePolicy
        </a>
         
        login.and
         
        <a
          href="https://memori.ai/en/tos"
          rel="noopener noreferrer"
          target="_blank"
        >
          login.termsOfService
        </a>
      </span>
    </label>
    <label
      class="memori-checkbox"
    >
      <span
        class="memori-checkbox--input-wrapper"
      >
        <input
          class="memori-checkbox--input"
          name="pAndCUAccepted"
          type="checkbox"
        />
        <span
          class="memori-checkbox--inner"
        />
      </span>
      <span
        class="memori-checkbox--text"
      >
        login.pAndCUAccepted
         
        <small>
          <em>
            (
            login.optional
            )
          </em>
        </small>
      </span>
    </label>
    <p>
      <small>
        login.deepThoughtExplaination
      </small>
    </p>
    <button
      class="memori-button memori-button--primary memori-button--rounded memori-button--padded"
      type="submit"
    >
      login.signUp
    </button>
    <p
      class="memori--login-drawer--signup"
    >
      login.alreadyHaveAnAccount
       
      <button
        class="memori-button memori-button--outlined memori-button--rounded memori-button--padded"
      >
        login.backToLogin
      </button>
    </p>
  </form>
</div>
`;
